packages/plugin-oas/src/SchemaGenerator.ts

```diff
@@ -560,6 +560,12 @@ export abstract class SchemaGenerator<
         type: 'type',
       })
 
+      const nullableEnum = schema.enum.includes(null);
+      if (nullableEnum) {
+        baseItems.push({ keyword: schemaKeywords.nullable })
+      }
+      const filteredValues = schema.enum.filter((value) => value !== null);
```

Comment:
This is the fix. If `null` is one of the enum members, then the whole enum should be consider nullable. But we don't want `null` as one of the members, since that's not how Typescript works, so I filter it out.